File: Dagger2Ex/app/src/main/java/expencive/vk/com/dagger2ex/car/PetroilEngine.java
package expencive.vk.com.dagger2ex.car;

import android.util.Log;

import javax.inject.Inject;

public class PetroilEngine implements Engine {

    private static final String TAG = "PetroilEngine";

    @Inject
    public PetroilEngine(){}

    @Override
    public void start() {

        Log.d(TAG, "start: Petroil Engine...............................");
        
    }
}

File: DaggerKotlin/app/src/main/java/vk/expencive/daggerkotlin/car/PetroilEngine.kt
package vk.expencive.daggerkotlin.car

import android.util.Log
import vk.expencive.daggerkotlin.car.Engine
import javax.inject.Inject




class  PetroilEngine @Inject constructor():
    Engine {
    private val TAG = "PetroilEngine"
    override fun wrom() {
        Log.d(TAG, "drive: Benzina rrrrrrrrrrrrrrrr ")
    }
}

